### PR TITLE
 SD-17: Indigo Visualization - add Donut chart visualization

### PR DESCRIPTION
**Pull request by naibinh**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1333/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1103/ : SUCCESS in 5m 04s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/576/ : SUCCESS in 3m 28s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1341/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1108/ : SUCCESS in 1m 43s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/581/ : SUCCESS in 2m 35s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1344/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1113/ : SUCCESS in 1m 10s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/586/ : SUCCESS in 2m 16s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/449/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1345/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/386/ : UNSTABLE in 9m 28s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/450/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1346/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/387/ : UNSTABLE in 8m 11s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1347/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1114/ : SUCCESS in 1m 19s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/587/ : SUCCESS in 2m 33s

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/451/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1348/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/388/ : UNSTABLE in 8m 08s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1350/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/389/ : FAILURE in 4m 51s

**Comment by gdgate** [comment]
Merge Failed.

This change or one of its cross-repo dependencies was unable to be automatically merged with the current state of its repository. Please rebase the change and upload a new patchset.

**Comment by naibinh** [comment]
extended test - storybook

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/454/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1354/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/391/ : UNSTABLE in 13m 55s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1387/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1145/ : SUCCESS in 1m 21s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/618/ : SUCCESS in 2m 35s

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/474/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1388/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/405/ : SUCCESS in 7m 45s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1415/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1165/ : FAILURE in 2m 04s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/638/ : SUCCESS in 2m 16s

**Comment by naibinh** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1435/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1183/ : SUCCESS in 3m 32s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/656/ : SUCCESS in 2m 13s

File: stories/DonutChart.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import { storiesOf } from '@storybook/react';
+import { screenshotWrap } from '@gooddata/test-storybook';
+
+import { PieChart } from '../src';
```
Comment (pbenes) [inline stories/DonutChart.tsx]: I think it should be reversed, i.e. importing { DonutChart } here (in DonutChart.tsx) and importing PieChart in PieChart stories below

File: stories/PieChart.tsx [hunk 2]
```diff
@@ -2,15 +2,15 @@ import * as React from 'react';
 import { storiesOf } from '@storybook/react';
 import { screenshotWrap } from '@gooddata/test-storybook';
 
-import { PieChart } from '../src';
+import { DonutChart } from '../src';
```
Comment (pbenes) [inline stories/PieChart.tsx]: I think PieChart is correct here and everywhere in this file (file is named PieChart.tsx)

File: src/components/DonutChart.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+export interface IDonutChartProps extends ICommonChartProps, IDonutChartBucketProps {
+    projectId: string;
+}
+
+type IDonutChartNonBucketProps = Subtract<IDonutChartProps, IDonutChartBucketProps>;
+
+/**
+ * [DonutChart](http://sdk.gooddata.com/gdc-ui-sdk-doc/docs/next/pie_chart_component.html)
```
Comment (ivan-nejezchleb) [inline src/components/DonutChart.tsx]: Don't forget update this link once Donut chart documentation is ready

Comment (naibinh) [reply src/components/DonutChart.tsx]: Noted.
Thank Ivan.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1458/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1201/ : SUCCESS in 1m 11s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/674/ : SUCCESS in 2m 16s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/493/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1460/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/422/ : UNSTABLE in 9m 32s

**Pull request closed without merging**